lib/giraph/remote/response.rb

```diff
@@ -0,0 +1,9 @@
+module Giraph
+  module Remote
+    # Dummy Hash-wrapper to enable precise `instance_of?` checks
+    # Allows us to differentiate JSON responses that are
+    # coming from remote GraphQL interface vs regular Hash objects
+    class Response < Hash
```

Comment:
Recursive nesting is needed because the JSON response is passed down the type tree for further resolution and this way each giraph'd-resolver can independently know what it needs to do (resolve on JSON or call original resolver) based on the type of Object in resolution without any global state.